File: examples/typescript-client/examples/quickstart.ts
/**
 * HotPlex Gateway - Quick Start Example
 * 
 * Minimal demo showing how to connect to the gateway and send a simple task.
 * 
 * Usage:
 *   npx tsx examples/quickstart.ts
 * 
 * Prerequisites:
 *   - HotPlex Gateway running at ws://localhost:8888/ws
 *   - Claude Code CLI installed and accessible
 */

import { HotPlexClient, WorkerType } from '../src/index.js';

async function main() {
  console.log('🚀 HotPlex Gateway - Quick Start\n');

  // Create client connecting to local gateway
  const client = new HotPlexClient({
    url: 'ws://localhost:8888/ws',
    workerType: WorkerType.ClaudeCode,
    apiKey: process.env.HOTPLEX_API_KEY || 'dev-api-key',
  });

  // Handle streaming output
  client.on('delta', (data) => {
    process.stdout.write(data.content);
  });

  // Handle completion
  client.on('done', (data) => {
    console.log('\n\n✅ Task completed:', data.success);
    if (data.stats) {
      console.log(`   Duration: ${data.stats.duration_ms}ms`);
      console.log(`   Tokens: ${data.stats.total_tokens}`);
      console.log(`   Cost: $${data.stats.cost_usd}`);
    }
    client.disconnect();
  });

  // Handle errors
  client.on('error', (data) => {
    console.error('\n❌ Error:', data.code, '-', data.message);
    client.disconnect();
    process.exit(1);
  });

  try {
    // Connect (creates new session)
    console.log('Connecting to gateway...');
    const ack = await client.connect();
    console.log(`Connected! Session: ${ack.session_id}\n`);
    console.log('Sending task to Claude Code...\n');

    // Send a simple task and wait for it to complete
    await client.sendInputAsync('Write a hello world program in Go that prints "Hello, World!" to stdout.');

    console.log('\nAll done!');

  } catch (err) {
    console.error('Task failed:', err instanceof Error ? err.message : err);
    client.disconnect();
    process.exit(1);
  }
}

main();

File: examples/java-client/src/main/java/dev/hotplex/example/QuickStart.java
package dev.hotplex.example;

import dev.hotplex.client.HotPlexClient;
import dev.hotplex.protocol.*;

import java.util.concurrent.CountDownLatch;
import java.util.concurrent.TimeUnit;

/**
 * HotPlex Gateway - Quick Start Example
 *
 * Minimal demo showing how to connect to the gateway and send a simple task.
 *
 * Usage:
 *   mvn compile && mvn exec:java -Dexec.mainClass="dev.hotplex.example.QuickStart"
 *
 * Environment Variables:
 *   HOTPLEX_GATEWAY_URL - Gateway URL (default: ws://localhost:8888)
 *   HOTPLEX_API_KEY     - Gateway API key (required)
 *   HOTPLEX_BOT_ID      - Bot ID for multi-bot isolation (optional)
 *   HOTPLEX_TASK        - Task to execute (optional)
 */
public class QuickStart {

    private static final String DEFAULT_GATEWAY_URL = "ws://localhost:8888";
    private static final String DEFAULT_TASK = "Write a hello world program in Go that prints \"Hello, World!\" to stdout.";

    public static void main(String[] args) throws Exception {
        System.out.println("🚀 HotPlex Gateway - Quick Start\n");

        // Configuration from environment
        String gatewayUrl = System.getenv("HOTPLEX_GATEWAY_URL");
        if (gatewayUrl == null || gatewayUrl.isEmpty()) {
            gatewayUrl = DEFAULT_GATEWAY_URL;
        }
        String apiKey = System.getenv("HOTPLEX_API_KEY");
        if (apiKey == null || apiKey.isEmpty()) {
            System.err.println("Error: HOTPLEX_API_KEY environment variable is required");
            System.err.println("Example: export HOTPLEX_API_KEY=your-api-key");
            System.exit(1);
        }
        String task = System.getenv("HOTPLEX_TASK");
        if (task == null || task.isEmpty()) {
            task = DEFAULT_TASK;
        }

        String botId = System.getenv("HOTPLEX_BOT_ID");

        // Create client using builder and use try-with-resources
        try (HotPlexClient client = HotPlexClient.builder()
                .url(gatewayUrl)
                .workerType("claude-code")
                .apiKey(apiKey)
                .botId(botId)
                .build()) {

            // Latch for keeping main thread alive until done
            CountDownLatch doneLatch = new CountDownLatch(1);

            // Set up event listeners
            client.on("messageDelta", (MessageDeltaData data) -> {
                // Streaming output
                if (data != null && data.getContent() != null) {
                    System.out.print(data.getContent());
                }
            });

            client.on("done", (DoneData data) -> {
                System.out.println("\n\n✅ Task completed!");

                if (data != null && data.getStats() != null) {
                    Object duration = data.getStats().get("duration_ms");
                    Object totalTokens = data.getStats().get("total_tokens");
                    Object cost = data.getStats().get("cost_usd");

                    System.out.println("   Duration: " + (duration != null ? duration + "ms" : "N/A"));
                    System.out.println("   Tokens: " + (totalTokens != null ? totalTokens : "N/A"));
                    System.out.println("   Cost: $" + (cost != null ? String.format("%.4f", ((Number) cost).doubleValue()) : "N/A"));
                }

                doneLatch.countDown();
            });

            client.on("error", (ErrorData data) -> {
                if (data != null) {
                    System.err.println("\n❌ Error: " + data.getCode() + " - " + data.getMessage());
                }
                doneLatch.countDown();
                System.exit(1);
            });

            client.on("disconnected", (String reason) -> {
                System.out.println("Disconnected: " + reason);
            });

            // Connect to gateway (creates new session)
            System.out.println("Connecting to gateway at: " + gatewayUrl);
            InitAckData ack = client.connect().get();
            System.out.println("Connected! Session: " + ack.getSessionId() + "\n");
            System.out.println("Sending task to Claude Code...\n");

            // Send input task
            client.sendInput(task);

            // Keep main thread alive until done or timeout
            boolean completed = doneLatch.await(5, TimeUnit.MINUTES);
            if (!completed) {
                System.out.println("\nTimeout waiting for task completion");
            }

        } catch (Exception e) {
            System.err.println("Failed to connect: " + e.getMessage());
            System.exit(1);
        }
        System.out.println("Disconnected.");
    }
}